library/pk_wrap.c

```diff
@@ -1095,6 +1095,103 @@ static int eckey_sign_rs_wrap(void *ctx, mbedtls_md_type_t md_alg,
 }
 #endif /* MBEDTLS_ECDSA_C && MBEDTLS_ECP_RESTARTABLE */
 
+#if !defined(ECP_FULL)
+/*
+ * Alternative function used to verify that the EC private/public key pair 
+ * is valid using PSA functions instead of ECP ones.
+ * The flow is:
+ * - sign a hash message using the provided private key
+ * - verify the signature using the public key
+ */
+static int eckey_alt_check_pair(const void *pub, const void *prv,
+                              int (*f_rng)(void *, unsigned char *, size_t),
+                              void *p_rng)
+{
+    (void)f_rng;
+    (void)p_rng;
+    psa_status_t status;
+    psa_key_attributes_t key_attr = PSA_KEY_ATTRIBUTES_INIT;
+    mbedtls_ecp_keypair *prv_ctx = (mbedtls_ecp_keypair *) prv;
+    mbedtls_ecp_keypair *pub_ctx = (mbedtls_ecp_keypair *) pub;
+    unsigned char sig[MBEDTLS_MPI_MAX_SIZE];
+    size_t sig_len = 0;
+    unsigned char hash[32];
+    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+    size_t curve_bits;
+    psa_ecc_family_t curve =
+                    mbedtls_ecc_group_to_psa(prv_ctx->grp.id, &curve_bits);
+    unsigned char key_buf[MBEDTLS_PSA_MAX_EC_KEY_PAIR_LENGTH];
+    size_t key_len = PSA_BITS_TO_BYTES(curve_bits);
+    mbedtls_svc_key_id_t key_id = MBEDTLS_SVC_KEY_ID_INIT;
+
+    memset(hash, 0x2a, sizeof(hash));
+
+    psa_set_key_type(&key_attr, PSA_KEY_TYPE_ECC_KEY_PAIR(curve));
+    psa_set_key_usage_flags(&key_attr, PSA_KEY_USAGE_SIGN_HASH);
+    // TODO: forcing SHA256 because this is included by default when building
```

Comment:
Ah, this is a problem with the "sign-verify" approach I suggested, indeed. In the meantime, I had another idea, inspired by what you did for key completion: we could just serialize the public key for both `prv` and `pub` (using `mbedtls_pk_write_pubkey()` I think) and see if the results are identical.

(Then later when #7202 is done, that would be `mbedtls_pk_write_pubkey()` on `prv` and directly compare to the byte array in `pub`, then when and its private counterpart is done, that would be `psa_import()` + `psa_export_public()` on `prv` and compare to byte array in `pub`.)

Wdyt?

Reply:
I like this idea!
However I have a question: since also for the key creation we are already importing the private key into PSA to get its public part, why can't we go that way directly also here? In theory also 7202 is a sort of middle step toward having everything on the PSA side (or at least there is a discussion on this)

Reply:
Let's continue this discussion in #7387.